pkg/api/v1/api.proto

```diff
@@ -17,6 +17,16 @@ option (gogoproto.goproto_unrecognized_all) = false;
 service CRIPluginService{
     // LoadImage loads a image into containerd.
     rpc LoadImage(LoadImageRequest) returns (LoadImageResponse) {}
+    // ResetPodSandbox resets the state of a sandbox and any containers that are part of the
+    // sandbox, recreating previously allocated resources (such as namespaces) that were freed
+    // on stop, and retaining other resources.
+    // This call is idempotent if the sandbox is in the ready state.
+    rpc ResetPodSandbox(ResetPodSandboxRequest) returns (ResetPodSandboxResponse) {}
+    // ResetContainer resets a stopped container back to the created state, keeping
+    // its scratch space untouched.
+    // This call is idempotent, and must not return an error if the container is already
+    // in the created state.
+    rpc ResetContainer(ResetContainerRequest) returns (ResetContainerResponse) {}
```

Comment:
Why do we need these new API calls if we can accomplish what we want in a call to `StartContainer` or `RunPodSandbox`?

Reply:
I didnt know if the orchestrator would, after restart, want to rest all the pods on the machine, and then start them, or restart them individually, so I added these to allow for as much control as possible, so they could decide how to handle different pods failing to restart.